Class/CleanupVMs.ps1

```diff
@@ -5,66 +5,99 @@ param
 
 $deleteVmBlock = {
-    Param ($ProfilePath, $vmName, $resourceId, $notVerbose)
-
+    Param ($ProfilePath, $vmName, $resourceId, $notVerbose, $HelperModule)
+    Import-Module $HelperModule
+    LogOutput "Deleting VM: $vmName"    
     if($notVerbose) {
         $ProgressPreference = "SilentlyContinue" # disable progress bar if not verbose
-    }
-    Write-Verbose "Deleting VM: $vmName"
+    }    
     Select-AzureRmProfile -Path $ProfilePath | Out-Null
     Remove-AzureRmResource -ResourceId $resourceId -ApiVersion 2016-05-15 -Force | Out-Null
-    Write-Verbose "Completed deleting $vmName"
+    LogOutput "Completed deleting $vmName"
 }
 
 # Iterate over all the VMs and delete any that we created
 foreach ($currentVm in $allVms){        
     $vmName = $currentVm.ResourceName
-    Write-Verbose "Starting job to delete VM $vmName"
-
-    $jobs += Start-Job -ScriptBlock $deleteVmBlock -ArgumentList $credentialPath, $vmName, $currentVm.ResourceId, $notVerbose   
+    LogOutput "Starting job to delete VM $vmName"
+    $jobs += Start-Job -Name $vmName -ScriptBlock $deleteVmBlock -ArgumentList $profilePath, $vmName, $currentVm.ResourceId, $notVerbose, $HelperModule
 }
 
-if($jobs.Count -ne 0)
-{
+$result = @{}
+$result.statusCode = "Not Started"
+$result.statusMessage = "Delete VMs process pending execution"
+$result.Succeeded = @()
+$result.Failed = @()
+
+if($jobs.Count -ne 0) {
     try{
-        Write-Verbose "Waiting for VM Delete jobs to complete"
+        $result.statusCode = "Started"
+        $result.statusMessage = "VM Delete jobs pending completion"
+        LogOutput "Waiting for VM Delete jobs to complete"
         foreach ($job in $jobs){
             Receive-Job $job -Wait -Force | Write-Verbose
         }
     } catch {
-        Write-Error “Caught an exception:” -ForegroundColor Red
-        Write-Error “Exception Type: $($_.Exception.GetType().FullName)” -ForegroundColor Red
-        Write-Error “Exception Message: $($_.Exception.Message)” -ForegroundColor Red                
+        LogError "Caught an exception:" -ForegroundColor Red
+        LogError "Exception Type: $($_.Exception.GetType().FullName)" -ForegroundColor Red
+        LogError "Exception Message: $($_.Exception.Message)" -ForegroundColor Red                
+        $result.statusCode = "Failed"        
+        $result.errorCode = $_.Exception.GetType().FullName
+        $result.statusMessage = $_.Exception.Message
     }
     finally{
-        Remove-Job -Job $jobs
+        Get-Job | ForEach-Object {
+            $status = @{}
+            $status.Name = $_.Name
+            $status.State = $_.State
+            if ($_.State -eq "Completed") {
+                $status.Details = "Job Succeeded"
+                $result.Succeeded += $status
+            } else {
+                $status.Details = (Get-Job -Name $_.Name).JobStateInfo.Reason
+                $result.Failed += $status
+            }
+        }
+        if ($result.Failed.Count -gt 0) {
```

Comment:
In theory this should execute only once after all the jobs have completed, but I don't think there is any harm in executing it in each iteration.